mpcite/doi_builder.py

```diff
@@ -0,0 +1,487 @@
+    uncited_materials
+
+    It will submit the uncited_materials to OSTI and constantly check whether OSTI has updated its database
+
+    if it finds one of the uncited_materials in OSTI, it will update the *dois* collection
+
+    From a high level, this Builder works as follows:
+    1. get_items -> get all the records that require registration / update
+    2. process_item -> compute the update to one particular data from get_items
+    3. update_targets -> batch update DOI collection and update remote services
+
+    Please note that the default run() function ties the above three functions together
+
+    """
+
+    def __init__(self,
+                 materials_store: Store,
+                 robocrys_store: Store,
+                 doi_store: Store,
+                 osti: OSTIModel,
+                 elsevier: ConnectionModel,
+                 max_doi_requests=1000,
+                 sync=True,
+                 **kwargs):
+        """
+         connection with materials database
+            1. establish connection with materials collection (Guaranteed online)
+            2. establish connection with doi collection (online or local)
+            3. establish connection with robocrys (Guaranteed online)
+
+        establish connection with ELink to submit info
+
+        establish connection with osti explorer to get bibtex
+
+
+        :param adapter: OstiMongoAdapter that keeps track of materials, doi, and other related stores
+        :param osti: osti connection specification
+        :param max_doi_requests: the maximum send size that E-Link allows
+        :param sync: boolean representing whether to sync DOI collection with Elink and Explorer in get_items
+        :param kwargs: other keywords fed into Builder(will be documented as development goes on)
+        """
+        super().__init__(sources=[materials_store, robocrys_store],
+                         targets=[doi_store],
+                         **kwargs)
+        # set connections
+        self.materials_store = materials_store
+        self.robocrys_store = robocrys_store
+        self.doi_store = doi_store
+        self.osti = osti
+        self.elsevier = elsevier
+        self.elink_adapter = ELinkAdapter(osti.elink)
+        self.explorer_adapter = ExplorerAdapter(osti.explorer)
+        self.elsevier_adapter = ElviserAdapter(self.elsevier)
+
+        # set flags
+        self.max_doi_requests = max_doi_requests
+        self.sync = sync
+
+        # set logging
+        self.logger.debug("DOI Builder Succesfully instantiated")
+
+    def get_items(self) -> Iterable:
+        """
+        Get a list of material that requires registration or update with remote servers
+
+        Steps:
+            1. Perform synchronization if sync is True
+                - Note that this is a ONE WAY sync, it only pulls data from remote server and checks whether local data
+                matches
+                - If a mismatch happened
```

Comment:
It just should just return the statemachine from the doi_store, but with the `valid` set to `False`